Libraries/Alert/Alert.js

```diff
@@ -17,7 +17,11 @@ export type AlertType =
   | 'plain-text'
   | 'secure-text'
   | 'login-password';
-export type AlertButtonStyle = 'default' | 'cancel' | 'destructive';
+export type AlertButtonStyle =
+  | 'default'
+  | 'cancel'
+  | 'destructive'
+  | 'preferred';
```

Comment:
I defo prefer Option 1 to Option 2, because 1 still allows you to make the Alert.alert() call and define the buttons together in one statement, whereas 2 requires you to create the Button objects in a different statement to pass the reference in 2 places.

The downside to 1 is that it's possible to represent an invalid case (multiple preferred buttons), but this API already has that problem (multiple cancel or destructive buttons), so at least we're locally consistent.

If I switch to Option 1, would we be ok to merge?

Reply:
I totally agree with the pros and cons you mentioned.

> If I switch to Option 1, would we be ok to merge?

Unfortunately, I have no permission to merge but I'm happy and welcome to go with Option 1 👍

Reply:
Hi @robbie-c, I really like this pull request. Do you have a chance to work on this?

Reply:
It's done